File: Universal Windows Platform Solution/Phantom of the West/Phantom of the West/Visual Novel/Serialization/GameState.cs
﻿using Phantom_of_the_West.Visual_Novel.Serialization.Event_Flagging;
using System.Runtime.Serialization;

namespace Phantom_of_the_West.Visual_Novel.Serialization
{
	[DataContract]
	public class GameState
	{
		[DataMember]
		internal EventFlagsCollection EventFlags
		{
			get;
			private set;
		}

		[DataMember]
		internal int ID
		{
			get;
			private set;
		}

		internal GameState(EventFlagsCollection eventFlags, int id)
		{
			EventFlags = eventFlags;
			ID = id;
		}
	}
}

File: Android Project/app/src/main/java/com/christopherraleigh/phantomofthewest/visual_novel/serialization/GameState.java
package com.christopherraleigh.phantomofthewest.visual_novel.serialization;

import com.christopherraleigh.phantomofthewest.visual_novel.serialization.event_flagging.EventFlagsCollection;

import java.io.Serializable;

/**
 * Created by christopherraleigh on 2016-04-01.
 */
public class GameState implements Serializable {
    private EventFlagsCollection eventFlags;

    private int id;

    public GameState(EventFlagsCollection eventFlags, int id) {
        this.eventFlags = eventFlags;
        this.id = id;
    }

    public EventFlagsCollection getEventFlags() {
        return eventFlags;
    }

    public int getID() {
        return id;
    }
}
